pkg/sql/opt/indexrec/index_recommendation_set.go

```diff
@@ -122,15 +186,98 @@ func (irs *IndexRecommendationSet) String() string {
 				if indexCol.Descending {
 					indexColSb.WriteString(" DESC")
 				}
+
 				indexCols[i] = indexColSb.String()
 			}
 
+			storingClause := ""
+			if indexRec.newStoredColOrds.Len() > 0 {
+				var storingSb strings.Builder
+				for i, col := range indexRec.newStoredColOrds.Ordered() {
```

Comment:
3 nested for loops and some conditionals is a lot of nesting. Maybe this can be broken up into helper functions?